pkg/tfvars/baremetal/baremetal.go

```diff
@@ -77,6 +79,7 @@ func TFVars(libvirtURI, apiVIP, imageCacheIP, bootstrapOSImage, externalBridge,
 			"power_interface":      accessDetails.PowerInterface(),
 			"raid_interface":       accessDetails.RAIDInterface(),
 			"vendor_interface":     accessDetails.VendorInterface(),
+			"deploy_interface":     "custom-agent",
```

Comment:
This should only be set if CustomDeploy is set.

Reply:
So, unless there is an installer interface to toggle the CustomDeploy value, we wont' know at this point whether we're using a custom-agent or something else. Also, won't this clash with the deploy_steps setting in the file?

Reply:
Okay, so I'm not sure if we need to support the old approach for new installations (openshift-installer is not used during upgrades, is it?). But it could be useful to have a switch, so that we can switch it off in case of regressions. This can be done in a follow-up.

Reply:
I think we should add the switch as a follow up as a BZ, if we see regressions. The changes might cascade down the stack to CAPBM, CBO, etc.

Reply:
Yes the installer is only used for initial deployment, not during upgrades, so IMO we should just switch to the new approach (provided we're confident with that)

Separately there is the issue of what happens on upgrade (where existing hosts and provisioning CR will exist referencing the qcow image etc), but that's outside the scope of any installer changes.

Reply:
Ok I thought of a problem after discussing this with @sadasu - some customers are using a customized qcow2 image as a workaround for missing day-1 networking support (e.g bond+vlans for the controlplane) - that essentially means we can't remove the qcow2 support until we have an alternative API for that configuration (or perhaps a way to achieve the same customization in this new flow, but I'd prefer to avoid another interim solution)

That probably means we should defer this until day-1 networking (which depends on this) is done, thoughts @kirankt @dtantsur @cybertron ?

Reply:
@hardys I agree. I think making changes this late and testing would take more time than 3 days. I am OK with closing this in favor of a more comprehensive work that supports qcow2 and live images.